### DISCUSSION TITLE
Improved detail in secondary import statement error message

### DISCUSSION THREAD

**Discussion by alvinj** [scala/scala3#17874] [Feature Requests]
When I incorrectly write code like this:

```
package test6

package t1
trait T1

package t2
trait T2
```

the compiler shows this error message:

```
-- [E040] Syntax Error: Test3.scala:7:0 ----------------------------------------
7 |trait T2
  |^^^^^
  |'{' expected, but 'end of statement' found
-- [E040] Syntax Error: Test3.scala:8:0 ----------------------------------------
8 |
  |^
  |'}' expected, but eof found
2 errors found
```

As I initially pointed out on Gitter, and Bjorn Regnell [clarifies here](https://gitter.im/lampepfl/dotty?at=603b66a6823b6654d273244a), the error message could be more helpful, especially when using the significant indentation syntax. Bjorn’s suggests this:

```
'{' or ':' expected, but 'end of statement' found
```

I think it would be even more clear to state that the `package` statement needs to end with a `:` and possibly that the `trait` line needs to be indented, but I don’t know what information is available. I’ve seen some other Scala 3 error messages that are really terrific and helpful — varargs, match expressions, trait inheritance, etc. — which is why I mention all of this.

**Comment by som-snytt** [comment]
Definitely a bug of some kind.

It's too late on a Sunday for me to understand that everything is in `p`:

```
  package p
class C
class X

package q:
  class D extends p.C
```

**Comment by bjornregnell** [comment]
I also think this is a bug rather than an enhancement, as the compiler should be able to be more helpful At least all "{ expected" should be changed to "{ or : expected" inline with the new syntactic options.
